tannerhodges/twitter-print-styles
# 🏥 Fix Unstable Selectors, Background Images, and More

## 🔬 How to Test

1. Download the ZIP: [chrome-twitter-print-styles-2.1.0-rc.1.zip](https://github.com/tannerhodges/chrome-twitter-print-styles/files/5165755/chrome-twitter-print-styles-2.1.0-rc.1.zip).
2. Unzip the folder.
3. Go to chrome://extensions/.
4. Click Load unpacked and select the unzipped folder.
5. Confirm the extension shows up under Extensions in the top right of your browser.
6. Go to twitter.com and test it on a few tweets.

<img width="1443" alt="Screen Shot 2020-09-02 at 10 28 23 PM" src="https://user-images.githubusercontent.com/1619485/92064599-c7f18800-ed6b-11ea-94a7-4a0c8be1b2e0.png">

<details><summary>Click here to see a GIF screencast of the new print window…</summary>

![chrome-twitter-print-styles-2 1 0-rc 1](https://user-images.githubusercontent.com/1619485/92064533-a7293280-ed6b-11ea-9a12-1f0c0154318f.gif)

</details>

## 🚧 Todo

- [x] Ask friends to test and confirm these fixes actually work.
- [x] Publish new version to Chrome Web Store.

---

## Added

- 👷‍♂️ Build Process FTW.

## Changed

- ✂️ Shorten document title below max filename length. _(Fixes #6)_
- 🖼 Open timeline in a separate window.
- 📷 Adjust media sizing to better fit within the page.
- 🏎 Add 50ms delay to help reduce risk of [rate limits](https://developer.twitter.com/en/docs/twitter-api/rate-limits).

## Fixed

- 🔥 Fix unstable timeline selector.
- 🙈 Fix background images not showing up on print. _(via [tweet](https://twitter.com/SeanTosi/status/1301279681136807940))_
- ⏰ Fix timestamp element not matching in other languages.
